### PR TITLE
Fix 368 by also matching `io.cucumber` annotations and interfaces

### PR DESCRIPTION
**Pull request by timtebeek**
As discussed in #368 ; First time developing on a plugin; Not tested yet but code seems straight forward. What's the flow needed to try this out?

### CODE REVIEW

**Comment by timtebeek** [comment]
Just tried a local `mvn clean install`, but I get
```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:1.3.0:compile (default-compile) on project cucumber.eclipse.editor: Compilation failure: Compilation failure: 
[ERROR] /home/tim/Documents/workspace/cucumber-eclipse/cucumber.eclipse.editor/src/main/java/cucumber/eclipse/editor/editors/GherkinKeywordsAssistProcessor.java:[21] 
[ERROR] 	import org.omg.PortableInterceptor.AdapterStateHelper;
[ERROR] 	       ^^^^^^^
```
So have not been able to test this yet. Any idea how to resolve? Do I need to update the binaries?

**Comment by qvdk** [comment]
> First time developing on a plugin

Great ! And welcome :)

> Pushed some code to a branch just now in #369 ; Could you have a look if this matches what you envisioned?

It's done. :+1: 

> Not tested yet but code seems straight forward. What's the flow needed to try this out?

Unfortunally, the plugin does not have many tests. Most of them are done manually. 
For now, I used the [examples from cucumber-jvm](https://github.com/cucumber/cucumber-jvm/tree/master/examples) to play with the plugin.

When I try the plugin manually, I generally play with
 - the java-calculator project from cucumber-jvm
 - a project where the step definitions are stored in an external jar file
 - and finally I play with it during some days on projects at work 

Since you introduce the support of the latest cucumber, the job should be done with the same samples with both cucmber v4 and v5.

> So have not been able to test this yet. Any idea how to resolve? Do I need to update the binaries?

Simply build it with JDK1.8 for now. 
Travis-CI built your branch successfully.

**Comment by timtebeek** [comment]
Hmm; tried it just now on the calculator examples, but it appears the new step definitions are not yet picked up. I also suddenly had to configure my projects with the Cucumber nature, which is also a first for me. Any idea what might be going on? My debug options/skills for a plugin a pretty limited.

File: cucumber.eclipse.steps.jdt/src/main/java/cucumber/eclipse/steps/jdt/JavaStepDefinitionsProvider.java [hunk 1]
```diff
@@ -103,12 +106,17 @@ protected static JavaStepDefinitionsProvider getInstance() {
 					}
 				}
 
-				// If import declaration matches with 'cucumber.api.java8'
-				// Then set Language of Java8-cuke-api
-				if (decl.getElementName().matches(REGEX_JAVA8_CUKEAPI)) {
-					String importDeclaration = decl.getElementName();
-					// setJava8CukeLang(importDeclaration);
```
Comment (timtebeek) [inline cucumber.eclipse.steps.jdt/src/main/java/cucumber/eclipse/steps/jdt/JavaStepDefinitionsProvider.java]: This code appeared left over from before; Hope it's OK to remove.

**Comment by timtebeek** [comment]
As discussed in slack, the testing issues might seem from having to match cucumber expressions in the latest calculator examples, rather than regular expressions in a previous iteration. Not confirmed yet, but something to be aware of in testing this branch.

**Comment by RusmanCool** [comment]
@timtebeek could you create a ZIP archive for a local installation? I have V 1.0.0.201907020018 that is not working.

**Comment by timtebeek** [comment]
@RusmanCool Unfortunately my changes are not working for me either; so a zip  won't help you much. If you know how to fix the code please do! I'm at a loss here, and don't have the time to dive in fully.

**Comment by sgrillon14** [comment]
Are there any ongoing actions to correct this problem? because this is blocking for all new version (io.cucumber.*)

**Comment by qvdk** [comment]
@timtebeek Thanks for your contribution !

@sgrillon14 You can use the snapshot update-site to get this fix.

**Comment by sgrillon14** [comment]
@qvdk , where is the delivery please?  i see https://marketplace.eclipse.org/content/cucumber-eclipse-plugin but last delivery is :

image

**Comment by sgrillon14** [comment]
I find https://github.com/cucumber/cucumber-eclipse-update-site-snapshot and it work update.

But plugin do not work

image

**Comment by qvdk** [comment]
I can not deliver to the Eclipse Marketplace, I do not have its access. @girijant could share it please?

The snapshot plugin works, but you need to setup your project from its contextual menu (right clic), `Configure` > `Convert as cucumber project...`

I added this trick in a [FAQ](https://github.com/cucumber/cucumber-eclipse/wiki/FAQ)

**Comment by sgrillon14** [comment]
it's already better, the plugin finds the steps that are in the project. I still have a problem, **it can't find the steps that are in a dependency (Maven)** that contains generic steps.

**Reproduce my problem:**

* run this Maven command: `mvn archetype:generate -B -DarchetypeGroupId=com.github.noraui -DarchetypeArtifactId=noraui-archetype -DarchetypeVersion=4.0.0.0 -DgroupId=com.your.company -DartifactId=robot -Dversion=0.0.1-SNAPSHOT -DinteractiveMode=false`

* Import robot project generated in Eclipse

* I added the cucumber nature to my project. Open the contextual menu from my project - right click - Configure > Convert as cucumber project... A cucumber icon will appears on my project.

* run project is OK but the plugin can't find the steps that are in a dependency (Maven)

image

**Pull request merged**